### ISSUE TITLE
Feature: Enhance `--action=parse --suspect-parses` to infer multiple ES6 exports.

### ISSUE THREAD

**Issue by ryan-roemer** [FormidableLabs/inspectpack#14]
of this form:

```js
/* 243 */
/*!************************************!*\
  !*** ./~/react-redux/lib/index.js ***!
  \************************************/
/***/ function(module, exports, __webpack_require__) {

	'use strict';
	
	exports.__esModule = true;
	exports.connect = exports.Provider = undefined;
	
	var _Provider = __webpack_require__(/*! ./components/Provider */ 244);
	
	var _Provider2 = _interopRequireDefault(_Provider);
	
	var _connect = __webpack_require__(/*! ./components/connect */ 246);
	
	var _connect2 = _interopRequireDefault(_connect);
	
	function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
	
	exports.Provider = _Provider2["default"];
	exports.connect = _connect2["default"];

/***/ }
```

Task:

Add a new `MULTIPLE_EXPORTS_ES` key with parse to detect this common form.

/cc @divmain

**Comment by divmain** [comment]
This would involve:
- Detect all `AssignmentExpression`s and `VariableDeclarator`s whose value is set by a call to `__webpack_require__`, and generate a map of identifier-to-webpack-module.
- Detect `CallExpressions` with a callee of `_interopRequireDefault` (we assume the Babel transform won't change the name of this function).
- When `_interopRequireDefault` `CallExpression` is detected in the context of an `AssignmentExpression` or `VariableDeclarator`, consult identifier-to-webpack-module map to determine which module is indirectly referenced.  When identified, add this new identifier to the identifier-to-webpack-module map.
- Use similar logic to what I added before to detect exports, referencing the identifier-to-webpack-module map.

That _should_ solve the case you've outlined above.  However, it would _not_ account for destructured imports/exports, which is probably equally likely to occur, and which could add significant complexity.  I can't remember how Babel transpiles those references - it may be `moduleName.exportName` which seems doable at first glance.

Keep in mind that adding a second parse-and-traverse step will worsen performance.  If you can detect as part of the same parse-and-traverse, that would be ideal.  However, if that's not an option, you may be able to at least cache that AST to avoid the second (or third) parse.

**Comment by ryan-roemer** [comment]
@divmain -- Maybe we just simplify the problem space to a heuristic of detecting just this instead:

```js
    exports.Provider = _Provider2["default"];
    exports.connect = _connect2["default"];
```

OR

```js
    exports.Provider = _Provider2.default;
    exports.connect = _connect2.default;
```

So basically:

```js
    exports.Foo = Foo.default;
    exports.Bar = Bar["default"];
```

How much effort would that take?

**Comment by divmain** [comment]
That would be _much_ easier, so long as 1) you won't need support for named imports, and 2) you don't care too much about the possibility of false-positives.  If that's the case, I can't imagine it would take more than a couple of hours.

**Comment by ryan-roemer** [comment]
Released in `inspectpack@0.4.1`
